components/skus/browser/skus_service_impl.cc

```diff
@@ -32,8 +32,14 @@ void OnFetchOrderCredentials(
     skus::FetchOrderCredentialsCallbackState* callback_state,
     skus::SkusResult result) {
   if (callback_state->cb) {
-    std::move(callback_state->cb).Run("");
+    std::string error_message;
+    if (result != skus::SkusResult::Ok) {
+      error_message = std::string{skus::result_to_string(result)};
+    }
+
+    std::move(callback_state->cb).Run(error_message);
```

Comment:
Once chromium adds support for `base::expected` in mojo we will probably want to use that here instead. Personally I would change the JS api to return an object that has both a status and an error message and pass two params here instead of using a non-empty string as an error condition. Maybe for a follow-up? In general using a non-empty string to indicate an error is highly discouraged

Reply:
The empty/non-empty string is especially problematic because some code uses it the opposite way: an empty string indicates an error. It's better to separate the result from any error message text.

Reply:
also if you define skus::SkusResult in mojom you can pass it directly to the callback